### ISSUE TITLE
slick.codegen.SourceCodeGenerator doesn't handle default value of TIMESTAMP

### ISSUE THREAD

**Issue by wentao** [slick/slick#1146]
In my sql statement, I use
       created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
To specify a column to use current timestamp as the default value.

Then if I run SourceCodeGenerator, it won't recognize the DEFAULT CURRENT_TIMESTAMP part, and not default value is provided in the case class in Tables.scala file.

I've also tried with "DEFAULT now()", it doesn't work as well.

**Comment by lucamolteni** [comment]
+1

**Comment by asdcdow** [comment]
+1

**Comment by szeiger** [comment]
Blocked on #213

**Comment by pocman** [comment]
I have a similar issue.
```
slick.SlickException: Could not parse default value Some(0000-00-00 00:00:00) for column schema.table.starts_at of type java.sql.Timestamp
```
```
Caused by: scala.MatchError: (0000-00-00 00:00:00,java.sql.Timestamp) (of class scala.Tuple2)
``` 
Issue is that my view is built using `Select ... Union ...` and I don't have the control over the default value (0000-00-00 00:00:00) inferred by Mysql.

**Comment by kuxuxun** [comment]
+1

**Comment by fysyihui** [comment]
+1

**Comment by jihuncho** [comment]
+1

**Comment by kentongray** [comment]
Same, " Could not parse default value Some(getdate()" would be nice if it would at least skip the default and move on.

**Comment by kentongray** [comment]
Looks like PR #1352 will help us suffering from this work around it hope it can make it to the next release! In the mean time you can basically take his changes and run it as your own entry point.

**Comment by guizmaii** [comment]
+1

**Comment by cvogt** [comment]
@kentongray @Falmarri or someone else mind submitting a PR setting the default for ignoring invalid defaults to true? (follow up to #1352)

**Comment by Sumith1896** [comment]
@cvogt Can I have a go at this? I may need some guidance though :)

**Comment by cvogt** [comment]
sure, whoever steps up first ;).   The task is super simple. Check the linked PR. ignoreInvalidDefaults is hardcoded to false in the main method. I'd suggest we hard code it to true. Also, the default values for the ignoreInvalidDefaults parameters on the methods should be true, not false.

**Comment by Sumith1896** [comment]
Thanks @cvogt. Will get it done.

**Comment by cvogt** [comment]
@kentongray so thanks to @Sumith1896, the default in master is now to skip and move on.

**Comment by cvogt** [comment]
For this ticket to be supported, we would need to support raw db values or functions as defaults. https://github.com/slick/slick/issues/214

**Comment by rlazar** [comment]
+1

**Comment by ntenisOT** [comment]
Is it possible to use the version after the merge without being released by Typesafe?

**Comment by cvogt** [comment]
@ntenisOT you can deploy locally, or I can push for a snapshot release among the maintainers.

**Comment by ntenisOT** [comment]
Can we have a snapshot release if that's not a big hassle for you?

**Comment by cvogt** [comment]
@ntenisOT ok, let me see

**Comment by ntenisOT** [comment]
Thanks a lot!

**Comment by ntenisOT** [comment]
I downloaded the repository and I run `sbt compile` in order to build the project and get the jars-snapshots. Where can I find the generated jars?Am I doing something completely wrong? I am new to SBT and Scala.

Thanks

**Comment by nafg** [comment]
Do publishLocal in SBT to install it in your local ivy repository, or
package to just get the jars.

**Comment by cvogt** [comment]
@ntenisOT yeah so `sbt publishLocal` deploys it to your local ivy cache, which means you can just use whatever version was deployed in your code as usual. The current version in master is `3.2.0-SNAPSHOT`.

**Comment by yukw777** [comment]
+1

**Comment by cvogt** [comment]
this might be tricky to fix. maybe @trevorsibanda and @radsaggi you can look at this one together later this summer. It might require changes in codegen and the drivers.

**Comment by trevorsibanda** [comment]
@cvogt I had already attempted this earlier. Made several changes to my first attempt and I'll open it as a seperate PR. @radsaggi We can look at it together and fix up some other stuff

**Comment by brianchizhang** [comment]
Any chance we can prioritize this? This is the one thing holding us back from utilizing CodeGen for Production usage. Much thanks

**Comment by cvogt** [comment]
@brianchizhang if you could revive #213 and see if it's working properly, I am sure @hvesalai is willing to release it.
